src/Platforms/DB2Platform.php

```diff
@@ -485,12 +495,12 @@ protected function getRenameIndexSQL(string $oldIndexName, Index $index, string
      */
     public function getDefaultValueDeclarationSQL(array $column): string
     {
-        if (! empty($column['autoincrement'])) {
+        if (isset($column['autoincrement']) && $column['autoincrement'] === true) {
             return '';
         }
 
-        if (! empty($column['version'])) {
-            if ((string) $column['type'] !== 'DateTime') {
```

Comment:
This might not be covered by tests, when I messed around and removed the `if` there was an error that Dbal/Types/DateTimeType cannot be converted to string, which makes sense because 'type' in column is a `DbalType` which doesn't have a `__toString()`, might need to be backported to 3.8 as well